### PR TITLE
Add build used by integrations-testing for Iceberg dependencies

### PR DESCRIPTION
**Pull request by snazy**
Builds on the changes for #4645 and adds a usually unused Gradle
build in the `iceberg/` directory. This new Gradle build will be
used by integrations-testing to build the Nessie projects that
have dependencies to Apache Iceberg.
    
Tools integrations testing will require a "surrounding" Gradle
build, which in turn includes three builds:
* Nessie build, but without all projects that depend on Apache
  Iceberg artifacts. For example: nessie-gc-base, spark-*
* Apache Iceberg build, which then uses the artifacts of the Nessie
  build.
* Nessie-Iceberg build containing the Gradle projects that are
  excluded in the first Nessie build, which can that way reference
  both the Nessie and Apache Iceberg artifacts. This includes the
  nessie-gc-base and nessie-spark-* projects.

### CODE REVIEW

**Comment by snazy** [comment]
The tools-integrations-tests project is currently located here: https://github.com/snazy/nessie-integration-tests
Testing workflow runs are [here](https://github.com/snazy/nessie-integration-tests/actions/workflows/main.yml).
Those runs do
* Build Nessie + Iceberg + Integrations-Tests using the latest versions of Nessie + Iceberg
* Run ITs in Nessie (nessie-spark-*-extensions), Nessie tests in Iceberg, ITs in the nessie-integration-tests. Note: the tests in nessie-integration-tests are very rudimentary.

The GH workflow to trigger the nessie-integration-tests isn't touched yet, because nessie-integration-tests is not yet under github.com/projectnessie and the branch in Nessie for the changes required for Iceberg is still in my personal fork.

Presto does currently not even build against latest Iceberg, so it's commented out.

File: nessie-iceberg/settings.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      mavenLocal()
+    }
+  }
+
+  plugins {
+
+    // Note: this is NOT a real project but a hack for dependabot to manage the plugin versions.
+    //
+    // Background: dependabot only manages dependencies (incl Gradle plugins) in build.gradle[.kts]
+    // files. It scans the root build.gradle[.kts] fila and those in submodules referenced in
```
Comment (dimas-b) [inline nessie-iceberg/settings.gradle.kts:46]: typo: `fila` -> `file`

**Review by dimas-b — COMMENTED** [review]
LGTM overall :+1: Minor comments below.

File: nessie-iceberg/settings.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+if (!JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_11)) {
+  throw GradleException("Build requires Java 11")
```
[the remark this answers is not in the document]
Comment (snazy) [reply nessie-iceberg/settings.gradle.kts:18]: Practically yes, but Spark related tests still require exactly Java 11. (Although Gradle pulls a Java 11 if it's not installed.)

**Review by dimas-b — APPROVED** [review]

**Pull request merged**